File: frontend/src/types/elevator.ts
export type RiskLevel = 'high' | 'medium' | 'low'

export interface Feature {
  name: string
  impact: number
  value: string
}

export interface ElevatorSummary {
  id: string
  building_name: string
  building_type: string
  floor_count: number
  model: string
  age_years: number
  risk_score: number
  risk_level: RiskLevel
  last_visit_date: string
  last_visit_technician: string
  in_model_scope: boolean
  zone: string
}

export interface ElevatorDetail extends ElevatorSummary {
  brand: string
  trend: number[]
  last_visit_notes: string
  nl_explanation: string
  features: Feature[]
  hourly_trips_avg: number
}

export interface PostVisitReport {
  technician_name: string
  visit_date: string
  failure_found: boolean
  components_replaced: string[]
  parameters_corrected: string[]
  notes: string
}

File: backend/app/schemas/elevator.py
from typing import Literal

from pydantic import BaseModel, ConfigDict


class FeatureSchema(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    name: str
    impact: float
    value: str
    direction: str


class ElevatorSummarySchema(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: str
    building_name: str
    building_type: str
    floor_count: int
    model: str
    age_years: int
    risk_score: float
    risk_level: Literal["high", "medium", "low"]
    last_visit_date: str
    last_visit_technician: str
    in_model_scope: bool
    zone: str


class ElevatorDetailSchema(ElevatorSummarySchema):
    brand: str
    trend: list[float]
    last_visit_notes: str
    nl_explanation: str
    features: list[FeatureSchema]
    hourly_trips_avg: int
